edgekv/examples/mpulse-currency-converter/main.js

```diff
@@ -0,0 +1,98 @@
+/*
+(c) Copyright 2020 Akamai Technologies, Inc. Licensed under Apache 2 license.
+Purpose:  Convert multiple currencies to one to support mPulse
+*/
+
+import { ReadableStream, WritableStream } from "streams";
+import { httpRequest } from "http-request";
+import { createResponse } from "create-response";
+import { TextEncoderStream, TextDecoderStream } from "text-encode-transform";
+import URLSearchParams from "url-search-params";
+import { EdgeKV } from "./edgekv.js";
+
+const edgeKv = new EdgeKV({ namespace: "currencies", group: "code" }); //replace namespace and group values with your own
+
+async function getCurrency(request) {
+  //in this example the currency code from the purchase order is pulled from a query parameter
+  let params = new URLSearchParams(request.query);
+  let currencyCode = params.get("currency");
+  let default_value = 1;
+  let currencyValue = 1;
+
+  currencyValue = await edgeKv.getJson({
+    item: currencyCode,
+    default_value: default_value,
+  });
+
+  return currencyValue;
+}
+
+export async function responseProvider(request) {
+  this.request = request;
```

Comment:
Not sure what this is achieving. Why do you believe it is needed?

Reply:
was probably remains from troubleshooting/collaboration with others, removed it